tests/test.py

```diff
@@ -107,6 +111,8 @@
 NoClientCert, RequestClientCert, RequireAnyClientCert, VerifyClientCertIfGiven, RequireAndVerifyClientCert = range(5)
 if TEST_WITH_TLS:
     ACRA_TLS_AUTH = RequireAndVerifyClientCert  # verify if provided https://golang.org/pkg/crypto/tls/#ClientAuthType
+    # Re-generate CRL so it is up-to-date when used in tests
+    os.system("SKIP_EXISTING=1 ./tests/ssl/generate_tls_keys.sh")
```

Comment:
By default, `tests/ssl/generate_tls_keys.sh` would re-generate all certs (and a CRL that contains revoked ones). So with `SKIP_EXISTING=1` we only ask it to create a fresh CRL and don't touch certs.
I've thought about creating a separate script only for generating that CRL.

Reply:
I mean, I don't understand why tests need to regenerate CRL. Isn't it already pregenerated and checked into the repository along with the certificates?

Reply:
The reason we need a "fresh" CRL because it contains date/time (called [Next Update](https://tools.ietf.org/html/rfc5280#section-5.1.2.5)) after which it is considered outdated, thus using such outdated CRL in tests leads to certificate denial and different errors, sometimes in unexpected places.
In `getCachedOrFetch()` there's a check `if crl.TBSCertList.NextUpdate.Before(time.Now()) { /* return error */ }` for this case.

Reply:
Oh... I did not know about that. Thank you for explanation!

I'm so obsessed with this because I like when the tests are more or less immutable and reproducible. Regenerating a part of them during CI runs might cause accidental breakage. Admittedly, certificates themselves might expire and thus stop working, but that is... um... circumvented by issuing them to be valid for the next 50 years.

I've looked around the `openssl ca` command line and it seems that `-crldays N` option allows to set the "next update" date for CRLs as well. I guess if this is set to 50 years in the future as well then it will not be necessary to update the CRL anytime sooner than the certificates it was generated from.